ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/diff/StateTraceResult.java

```diff
@@ -0,0 +1,410 @@
+        // Skip if all pre-state fields are empty
+        return;
+      }
+
+      gen.writeObjectFieldStart(addr);
+
+      // Balance
+      accountDiff
+          .getBalance()
+          .getFrom()
+          .ifPresent(
+              b -> {
+                try {
+                  gen.writeStringField("balance", b);
+                } catch (IOException e) {
+                  throw new RuntimeException(e);
+                }
+              });
+
+      // Code (skip if empty bytecode)
+      accountDiff
+          .getCode()
+          .getFrom()
+          .ifPresent(
+              code -> {
+                if (!Bytes.fromHexString(code).isEmpty()) {
+                  try {
+                    gen.writeStringField("code", code);
+                  } catch (IOException e) {
+                    throw new RuntimeException(e);
+                  }
+                }
+              });
+
+      // Code hash (skip if empty hash)
+      accountDiff
+          .getCodeHash()
+          .getFrom()
+          .ifPresent(
+              codeHash -> {
+                if (!Hash.EMPTY.equals(Hash.fromHexString(codeHash))) {
+                  try {
+                    gen.writeStringField("codeHash", codeHash);
+                  } catch (IOException e) {
+                    throw new RuntimeException(e);
+                  }
+                }
+              });
+
+      // Nonce (skip if zero)
+      if (accountDiff.getNonce().getFrom().isPresent()) {
+        String nonceStr = accountDiff.getNonce().getFrom().get();
+        try {
+          long nonce = Long.decode(nonceStr);
+          if (nonce != 0) {
+            gen.writeNumberField("nonce", nonce);
+          }
+        } catch (NumberFormatException e) {
+          gen.writeStringField("nonce", nonceStr);
+        }
+      }
+
+      // Storage entries
+      var storageEntries = accountDiff.getStorage().entrySet().stream().toList();
+      if (!storageEntries.isEmpty()) {
+        gen.writeObjectFieldStart("storage");
+        for (var se : storageEntries) {
+          DiffNode node = se.getValue();
+          gen.writeStringField(se.getKey(), node.getFrom().get());
+        }
+        gen.writeEndObject();
+      }
+      gen.writeEndObject();
+    }
+
+    /** Writes the "pre" state for an account (diff mode). Only includes fields that differ. */
+    private void writePreNode(
+        final JsonGenerator gen, final String addr, final AccountDiff accountDiff)
+        throws IOException {
+
+      if (!accountDiff.hasDifference() || shouldSkipNode(accountDiff)) {
+        return;
+      }
+
+      gen.writeObjectFieldStart(addr);
+
+      // Balance, Code, CodeHash, Nonce (same logic as writeNode)
+      accountDiff
+          .getBalance()
+          .getFrom()
+          .ifPresent(
+              b -> {
+                try {
+                  gen.writeStringField("balance", b);
+                } catch (IOException e) {
+                  throw new RuntimeException(e);
```

Comment:
It probably makes sense to fail early if any exception occurs, but the same isn't being done for fields in `writePostNode`, should it be?